File: app/src/main/java/com/example/dcct/model/Impl/SignOutModelImp.java
package com.example.dcct.model.Impl;

import android.util.Log;

import com.example.dcct.base.BaseObserver;
import com.example.dcct.internet.InternetAPI;
import com.example.dcct.bean.BackResultData;
import com.example.dcct.model.SignOutModel;
import com.example.dcct.internet.NetWorkApi;
import com.example.dcct.view.SignOutCallback;

import java.util.Objects;

import io.reactivex.Observable;

public class SignOutModelImp implements SignOutModel {
    private static final String TAG = "SignOutPresenterImp";

    @Override
    public void postSignOutId(long uid,ReturnSignState returnSignState) {
        InternetAPI internetApi = NetWorkApi.getInstance().getService();
        Observable<BackResultData> compose = internetApi.subSignOutId( uid )
                .compose( NetWorkApi.applySchedulers( new BaseObserver<BackResultData>() {
                    @Override
                    public void onSuccess(BackResultData backResultData) {
                        Log.d( TAG, "退出登录 ==》" + backResultData.toString() );
                        returnSignState.onComplete( backResultData );
                    }

                    @Override
                    public void onFailure(Throwable e) {
                        returnSignState.onError( e.getMessage() );
                    }
                } ) );
    }
}

File: app/src/main/java/com/example/dcct/model/impl/SignOutModelImp.kt
package com.example.dcct.model.impl

import com.example.dcct.base.BaseObserver
import com.example.dcct.bean.BackResultData
import com.example.dcct.internet.NetWorkApi.Companion.applySchedulers
import com.example.dcct.internet.NetWorkApi.Companion.instance
import com.example.dcct.model.SignOutModel
import com.example.dcct.model.SignOutModel.ReturnSignState

class SignOutModelImp : SignOutModel {
    override fun postSignOutId(uid: Long, returnSignState: ReturnSignState) {
        val internetApi = instance!!.service
        val compose = internetApi.subSignOutId(uid)
                .compose(applySchedulers(object : BaseObserver<BackResultData<*>>() {
                    override fun onSuccess(t: BackResultData<*>) {
//                        Log.d( TAG, "退出登录 ==》" + backResultData.toString() );
                        returnSignState.onComplete(t)
                    }

                    override fun onFailure(e: Throwable) {
                        e.message?.let { returnSignState.onError(it) }
                    }
                }))
    }

    companion object {
        private const val TAG = "SignOutPresenterImp"
    }
}